RocketChat/Rocket.Chat.ReactNative
# feat: mobile troubleshoot notifications

## Proposed changes

Problem statement:
- Push notifications are the top cause of bad reviews in both Google Play and App Store.

Proposed solution: 
- Create a troubleshooting section in the app to help users identify the possible issues related to push notifications


As we aren't merging the Push Quota into this PR, I removed the feature of it from here and added at this branch https://github.com/RocketChat/Rocket.Chat.ReactNative/tree/feat.troubleshooting-notification-push-quota. When the Push Quota will be done on the backend side, we could work with that branch.

## Issue(s)	


## How to test or reproduce


- The user can go to Troubleshoot View from:
   - Rooms List View
   - Room View (when is in alert)
   - Room View (clicking in the icon when the icon is not in alert) -> Notification Preference -> Troubleshoot View
   - Drawer (click at hamburger icon) -> Profile -> Notifications -> Troubleshoot View

- The `Device notification settings` refers the permission to receive notification in the device, if you enable it or not
- The push Gateway connection will appear only for servers equal to or greater than 6.6.0
- There are 3 values for push gateway connection:
   -  the green icon -> when uses the default gateway
   - the orange icon -> when the servers uses a custom gateway
   - the red icon -> when the gateway isn't connected
- To `test the push notification` there is a permission `'test-push-notifications'`, then the workspace's admin can determine who is able to test the push-notification, as default of the server all the users can test it. If the user isn't able to test, the button should be disabled.

## Screenshots


https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/64a45a5a-ee33-418e-94cd-340c7497ae48


## Types of changes




## Checklist



## Further comments
